File: Proyecto_backend/src/flaskr/dtos/create_order_request_dto.py
from flaskr.dtos.base_dto import BaseDTO

class CreateOrderRequestDTO(BaseDTO):
    def get_required_fields(self):
        return {
            'user_id': int,
            'total': float,
            'status': str,
            'order_items': list  # This will be a list of OrderHasMenuDTO
        }

    def get_field_constraints(self):
        return {
            'status': {'min_length': 1, 'max_length': 20},
            'total': {'min_value': 0},
        }

File: Proyecto_frontend/js/dtos/CreateOrderRequestDto.js
class CreateOrderRequestDto {
    constructor(user_id, total, status, order_items) {
        this.user_id = user_id;
        this.total = total;
        this.status = status;
        this.order_items = order_items; // This should be a list of OrderHasMenuDTO
    }
}


export default CreateOrderRequestDto;
